tests/NuGet.IndexingTests/DownloadsByVersionTests.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using NuGet.Indexing;
+using Xunit;
+
+namespace NuGet.IndexingTests
+{
+    public class DownloadsByVersionTests
+    {
+        [Theory]
+        [MemberData(nameof(RollingTotalTestData))]
+        public void RollingTotalCountTest(Dictionary<string, int> versionDownloads, int expectedTotal)
+        {
+            var newDownloadsByVersion = new DownloadsByVersion();
+            var rollingTotal = 0;
+            foreach (var version in versionDownloads)
+            {
+                newDownloadsByVersion[version.Key] = version.Value;
+                rollingTotal += version.Value;
+                Assert.Equal(rollingTotal, newDownloadsByVersion.Total);
+                Assert.Equal(newDownloadsByVersion[version.Key], version.Value);
+            }
+
+            Assert.Equal(expectedTotal, newDownloadsByVersion.Total);
+        }
+
+        [Theory]
+        [MemberData(nameof(UpdateTotalTestData))]
+        public void UpdateTotalTest(Dictionary<string, int> originalData, Dictionary<string, int> updatedVersionDownloads, int expectedTotal)
+        {
+            var newDownloadsByVersion = new DownloadsByVersion();
+            foreach (var entry in originalData)
+            {
+                newDownloadsByVersion[entry.Key] = entry.Value;
+            }
+
+            foreach (var version in updatedVersionDownloads)
+            {
+                newDownloadsByVersion[version.Key] = version.Value;
+            }
+
+            Assert.Equal(expectedTotal, newDownloadsByVersion.Total);
+        }
+
+        [Fact]
+        public void ReturnZeroForUnknownTest()
+        {
+            var newDownloadsByVersion = new DownloadsByVersion();
+            Assert.Equal(0, newDownloadsByVersion["not.real.version"]);
+        }
+
+        public static IEnumerable<object[]> RollingTotalTestData
```

Comment:
I think this code would be a bit more readable if you moved the test data to be closer to the tests themselves rather than grouping them together.

E.g.
```
public static IEnumerable<object[]> RollingTotalTestData
{
    ...
}

public void RollingTotalCountTest(Dictionary<string, int> versionDownloads, int expectedTotal)
{
    ...
}

public static IEnumerable<object[]> UpdateTotalTestData
{
    ...
}

public void UpdateTotalTest(Dictionary<string, int> originalData, Dictionary<string, int> updatedVersionDownloads, int expectedTotal)
{
    ...
}
```